pkg/validator.go

```diff
@@ -0,0 +1,74 @@
+package pkg
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+)
+
+
+type QuotaValidator struct {
+	RulesContainer *RulesContainer
+}
+
+func (qv *QuotaValidator) Handle(quota string,history OrderHistory, newOrder string ) error {
+	//todo: move to map number of products increase
+	switch quota {
+		case BASIC: return qv.onBasic(history,newOrder)
+		case MANIAC: return qv.onManiac(history,newOrder)
+		case LOVER: return qv.onLover(history,newOrder)
+	}
+	
+	return errors.New("no such quota: " + quota)
+}
+
+func (qv *QuotaValidator) onBasic(history OrderHistory, newOrder string) error {
```

Comment:
having functions that are tighten to config-related entities, is not good.
If we introduce one more membership, we'll need to add a new function (but it should be just changing the config)

Reply:
fixed